### PR TITLE
Update Version to 4.2.4.0.0.1

### PR DESCRIPTION
**Pull request by helium-service**
Adapter version '4.2.4.0.0.1', partner version '~> 2.4.0.0'

### CODE REVIEW

File: Source/BidMachineAdapter.swift [hunk 1]
```diff
@@ -142,7 +168,8 @@ final class BidMachineAdapter: PartnerAdapter {
         case .banner:
             return BidMachineAdapterBannerAd(adapter: self, request: request, delegate: delegate)
         default:
-            // Not using the `.adaptiveBanner` case directly to maintain backward compatibility with Chartboost Mediation 4.0
+            // Not using the `.adaptiveBanner` or `.rewardedInterstitial cases directly to maintain
```
Comment (CB-HaoxinLi) [inline Source/BidMachineAdapter.swift:171]: nit: `.rewardedInterstitial needs a trailing `

File: Source/BidMachineAdapter.swift [hunk 2]
```diff
@@ -72,14 +73,39 @@ final class BidMachineAdapter: PartnerAdapter {
...
+            } else {
+                let error = error(.prebidFailureInvalidArgument, description: "Unsupported ad format")
+                log(.fetchBidderInfoFailed(request, error: error))
+                completion(nil)
+                return
+            }
+        }
+
+        BidMachineSdk.shared.token(with: placementFormat) { [self] token in
+            guard let token = token else {
```
Comment (CB-HaoxinLi) [inline Source/BidMachineAdapter.swift]: nit: I believe we can simply `guard let token else` now

**Review by CB-HaoxinLi — COMMENTED** [review]
LGTM with two nit comments. Thanks!

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,5 +3,8 @@
 Note the first digit of every adapter version corresponds to the major version of the Chartboost Mediation SDK compatible with that adapter. 
 Adapters are compatible with any Chartboost Mediation SDK version within that major version.
 
+### 4.2.4.0.0.1
+- This version of the adapter has been certified with BidMachine 2.4.0.0.
```
Comment (daniel-barros) [inline CHANGELOG.md:8]: Can we indicate what's changed from the previous version?

**Comment by ricealexanderb** [comment]
@CB-HaoxinLi, @daniel-barros I've made all the changes, thanks for catching those.

**Review by daniel-barros — APPROVED** [review]

**Comment by CB-HaoxinLi** [comment]
The latest commit LGTM 👍

**Pull request merged**